File: shared/domain/stock.py
"""Core stock and market domain models."""

from __future__ import annotations

from datetime import date
from typing import Literal, Self

from pydantic import BaseModel, model_validator


class Stock(BaseModel):
    code: str
    name: str
    market: Literal["KOSPI", "KOSDAQ"]
    sector: str | None
    industry: str | None
    listed_at: date
    delisted_at: date | None
    is_delisted: bool = False

    @model_validator(mode="after")
    def derive_is_delisted(self) -> Self:
        self.is_delisted = self.delisted_at is not None
        return self

File: app/lib/domain/entities/stock.dart
/// File: app/lib/domain/entities/stock.dart
///
/// Role:
///   Dart mirror of `shared.domain.stock.Stock` (Python Pydantic).
///   Immutable value object representing one tradeable security.
///
/// Fields (planned):
///   - code (e.g. "005930")
///   - name
///   - market (KOSPI | KOSDAQ)
///   - sector, industry
///   - listedAt, delistedAt
///   - isDelisted
///
/// Connected modules:
///   - data/api/*  (deserialize from JSON)
///   - presentation/* (every screen that shows a stock)
